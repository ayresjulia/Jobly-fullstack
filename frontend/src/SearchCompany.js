import React from "react";
import { Redirect, useParams, Link } from "react-router-dom";
import { Card, CardBody, CardTitle, CardSubtitle } from "reactstrap";
import "./SearchCompany.css";

/* Single company card. */

const SearchCompany = ({ companies, jobs, cantFind }) => {
	const { handle } = useParams();

	let company = companies.find((company) => company.handle === handle);

	// filter jobs according to company handle to display on company detail page
	let job = jobs.filter((job) => job.companyHandle === company.handle);

	if (!company) return <Redirect to={cantFind} />;

	return (
		<section>
			<Card className="Company">
				<CardBody>
					<CardTitle className="font-weight-bold text-center">{company.name}</CardTitle>
					<CardSubtitle tag="h6" className="mb-2 text-muted text-center">
						{company.description}
					</CardSubtitle>

					<b>Jobs available: </b>
					<ul>
						{job.map((j) => (
							<li key={j.id}>
								<Link to={`/jobs/${j.id}`} key={j.id}>
									<i>{j.title}</i>{" "}
								</Link>
							</li>
						))}
					</ul>
				</CardBody>
			</Card>
		</section>
	);
};

export default SearchCompany;
