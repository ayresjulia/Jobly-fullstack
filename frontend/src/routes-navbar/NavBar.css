.navbar-welcome {
	font-size: 25px;
}
.navbar-link {
	margin-right: 20px;
	font-size: 18px;
}
