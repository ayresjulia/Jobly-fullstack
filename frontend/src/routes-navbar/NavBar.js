import React from "react";
import "./NavBar.css";
import { NavLink } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";

/** Navigation bar that changes according to currentUser. */

const NavBar = ({ currentUser, logout }) => {
	return (
		<div>
			<Navbar expand="md" color="light">
				<NavLink exact to="/" className="navbar-welcome">
					Jobly
				</NavLink>
				{currentUser && (
					<Nav className="ml-auto" navbar>
						<NavItem className="navbar-link">
							<NavLink to="/companies">Companies</NavLink>
						</NavItem>
						<NavItem className="navbar-link">
							<NavLink to="/jobs">Jobs</NavLink>
						</NavItem>
						<NavItem className="navbar-link">
							<NavLink to="/profile">Profile</NavLink>
						</NavItem>
						<NavItem className="navbar-link">
							<NavLink to="/" onClick={logout}>
								Log Out
							</NavLink>
						</NavItem>
					</Nav>
				)}
				{!currentUser && (
					<Nav className="ml-auto" navbar>
						<NavItem className="navbar-link">
							<NavLink to="/login">Log In</NavLink>
						</NavItem>
						<NavItem className="navbar-link">
							<NavLink to="/signup">Sign Up</NavLink>
						</NavItem>
					</Nav>
				)}
			</Navbar>
		</div>
	);
};

export default NavBar;
