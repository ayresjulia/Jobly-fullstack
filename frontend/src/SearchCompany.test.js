import React from "react";
import { render } from "@testing-library/react";
import SearchCompany from "./SearchCompany";
import { MemoryRouter } from "react-router";
import NavBar from "./routes-navbar/NavBar";

let currentUser = "testuser";

beforeEach(() => {
	currentUser;
});

afterEach(() => {
	currentUser = "";
});

it("renders without crashing", () => {
	render(
		<MemoryRouter>
			<NavBar currentUser={currentUser}>
				<SearchCompany />
			</NavBar>
		</MemoryRouter>
	);
});

it("matches snapshot", () => {
	const { asFragment } = render(
		<MemoryRouter>
			<NavBar currentUser={currentUser}>
				<SearchCompany />
			</NavBar>
		</MemoryRouter>
	);
	expect(asFragment()).toMatchSnapshot();
});
