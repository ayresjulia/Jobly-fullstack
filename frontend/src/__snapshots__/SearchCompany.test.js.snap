// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md bg-light"
    >
      <a
        aria-current="page"
        class="navbar-welcome active"
        href="/"
      >
        Jobly
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="navbar-link nav-item"
        >
          <a
            href="/login"
          >
            Log In
          </a>
        </li>
        <li
          class="navbar-link nav-item"
        >
          <a
            href="/signup"
          >
            Sign Up
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
